app/models/call_session.rb

```diff
@@ -1,33 +1,53 @@
 class CallSession
-  KEY = 'CALL-SESSION'
   EXPIRE_CALL_SESSION_IN = 24.hours
+  COUNTRY = 'IN'
 
-  def initialize(user_phone_number, patient_phone_number)
-    @user = User.find_by_phone_number(user_phone_number)
+  attr_reader :patient_phone_number, :user
+
+  def initialize(call_id, user_phone_number, patient_phone_number)
+    @call_id = call_id
+    @user = User.find_by_phone_number(sanitized_phone_number(user_phone_number))
     @patient_phone_number = PatientPhoneNumber.find_by_number(patient_phone_number)
   end
 
-  def save(call_id)
-    Rails.cache.write(session_key(call_id),
+  def authorized?
+    @user.present? &&
```

Comment:
If a user is not approved to sync, can they still make calls? 
`authorized?` will currently allow that, we should check if this is okay.

Reply:
This is a good point. We'll add this in.